cpp/src/neighbors/cagra_c.cpp

```diff
@@ -297,6 +337,68 @@ void* _merge(cuvsResources_t res,
 
   return merged_index;
 }
+
+template <typename output_mdspan_type, typename T, typename IdxT>
+void _get_dataset(cuvsResources_t res, cuvsCagraIndex_t index, DLManagedTensor* dataset)
+{
+  auto res_ptr   = reinterpret_cast<raft::resources*>(res);
+  auto index_ptr = reinterpret_cast<cuvs::neighbors::cagra::index<T, IdxT>*>(index->addr);
+  auto dst       = cuvs::core::from_dlpack<output_mdspan_type>(dataset);
+  auto src       = index_ptr->dataset();
+
+  RAFT_EXPECTS(src.extent(0) == dst.extent(0), "Output dataset has incorrect number of rows");
+  RAFT_EXPECTS(src.extent(1) == dst.extent(1), "Output dataset has incorrect number of cols");
+
+  // dataset from cagra could be strided, account for that when copying over
+  raft::copy_matrix(dst.data_handle(),
```

Comment:
Does this mean that the `dataset` is copied unconditionally to `__host__` memory?  Does the recent discussion regarding returning a `__device__` memory reference have bearing on this bit?

Reply:
@benfred and I spoke offline about this. The current plan is to have the C API populate the given dltensor and just reuse the existing device pointer (from the index).

Reply:
I've added this in  https://github.com/rapidsai/cuvs/pull/1086/commits/2f22d5356657dabd810ec8a74d36980b02f286ee - basically returns a DLTensor that references the device memory from the cagra index itself without copying.